app/models/observation_accuracy_experiment.rb

```diff
@@ -586,16 +431,224 @@ def get_sample_for_user_id( user_id )
     )
   end
 
-  def get_incorrect_observations_in_sample
-    samples = ObservationAccuracySample.where( observation_accuracy_experiment_id: id, correct: 0 )
-    return nil unless samples.count.positive?
+  def get_top_level_stats( subset )
+    subset_filter = case subset
+    when "verifiable_results" then "( quality_grade = 'research' OR quality_grade = 'needs_id' )"
+    when "all_results" then nil
+    else "quality_grade = 'research'"
+    end
 
-    oids = samples.map( &:observation_id ).join( "," )
-    FakeView.identify_observations_url(
-      place_id: "any",
-      reviewed: "any",
-      quality_grade: "needs_id,research,casual",
-      id: oids
-    )
+    counts = {
+      incorrect: observation_accuracy_samples.
+        where( [subset_filter, "correct = 0"].compact.join( " AND " ).to_s ).count,
```

Comment:
instead of doing a compact and join here, this can just chain multiple `.where` together. e.g. `observation_accuracy_samples.where( subset_filter ).where( "correct = 0" ).count`. When subset_filter is nil, adding a `.where( nil )` is just ignored. Alternatively, you can set up a scope at the top of the method and append to it in the case switch before querying. Something like this: 
```
    counts_base_scope = observation_accuracy_samples
    case subset
    when "verifiable_results"
      counts_base_scope = counts_base_scope.where( "quality_grade = 'research' OR quality_grade = 'needs_id'" )
    when "all_results"
      # do nothing
    else
      counts_base_scope = counts_base_scope.where( "quality_grade = 'research'" )
    end

    counts = {
      incorrect: counts_base_scope.where( "correct = 0" ).count,
      uncertain: counts_base_scope.where( "correct IS NULL OR correct = -1" ).count,
      correct: counts_base_scope.where( "correct = 1" ).count
    }
```